win32/tests/file_win32_ut/file_win32_ut.c

```diff
@@ -0,0 +1,952 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdbool.h>
+#endif
+#include "windows.h"
+#include "azure_macro_utils/macro_utils.h"
+
+void* real_malloc(size_t size)
+{
+    return malloc(size);
+}
+
+void real_free(void* ptr)
+{
+    free(ptr);
+}
+
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_windows.h"
+#include "umock_c/umocktypes_bool.h"
+#include "umock_c/umock_c_negative_tests.h"
+
+
+#define ENABLE_MOCKS
+
+#include "azure_c_pal/execution_engine.h"
+#include "azure_c_pal/gballoc.h"
+#include "execution_engine_win32.h"
+#include "mock_file.h"
+
+MOCKABLE_FUNCTION(, void, mock_user_callback, void*, user_context, bool, is_successful);
+
+#undef ENABLE_MOCKS
+
+#include "azure_c_pal/file.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+TEST_DEFINE_ENUM_TYPE(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_RESULT)
+IMPLEMENT_UMOCK_C_ENUM_TYPE(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_VALUES)
+
+TEST_DEFINE_ENUM_TYPE(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_RESULT)
+IMPLEMENT_UMOCK_C_ENUM_TYPE(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+static FILE_HANDLE get_file_handle(const char* filename, PTP_WIN32_IO_CALLBACK* captured_callback)
+{
+    ///arrange

```

Comment:
>///arrange [](start = 4, length = 10)

You can remove arrange/act/assert from the helper functions, we really just use that convention for the tests themselves